app/src/main/java/net/gsantner/markor/util/BackupRestoreHelper.java

```diff
@@ -0,0 +1,211 @@
+            SearchReplaceDialog.SEARCH_REPLACE_SETTINGS
+    };
+
+    private static final Pattern[] PREF_EXCLUDE_PATTERNS = {
+            Pattern.compile("^pref_key__.*encryption_password.*", Pattern.MULTILINE),
+    };
+
+    public static void backupConfig(final Context context, final FragmentManager manager) {
+
+        if (context instanceof Activity) {
+            final Activity activity = (Activity) context;
+
+            FilesystemViewerCreator.showFolderDialog(
+                    new FilesystemViewerData.SelectionListenerAdapter() {
+                        @Override
+                        public void onFsViewerConfig(FilesystemViewerData.Options dopt) {
+                            dopt.rootFolder = new AppSettings(context).getNotebookDirectory();
+                            dopt.titleText = R.string.select_folder;
+                        }
+
+                        @Override
+                        public void onFsViewerSelected(String request, File dir) {
+                            createAndSaveBackup(context, getSaveFile(dir));
+                        }
+                    }, manager, activity
+            );
+        }
+    }
+
+    public static File getSaveFile(final File folder) {
+        File file;
+        int index = 0;
+        do {
+            file = new File(folder, String.format(SAVE_NAME, (index == 0) ? "" : String.format("_%d", index)));
+            index++;
+        } while (file.exists());
+        return file;
+    }
+
+    public static String getPrefName(final Context context, final String raw) {
+        if (TextUtils.isEmpty(raw)) {
+            return context.getPackageName() + "_preferences";
+        } else {
+            return raw;
+        }
+    }
+
+    public static boolean includeKey(final String key) {
+        for (final Pattern ep : PREF_EXCLUDE_PATTERNS) {
+            if (ep.matcher(key).matches()) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public static void createAndSaveBackup(final Context context, final File saveLoc) {
+        try {
+            final JSONObject json = new JSONObject();
+            for (final String _pref : PREF_NAMES) {
+                final String pref = getPrefName(context, _pref);
+                final SharedPreferences sp = context.getSharedPreferences(pref, Context.MODE_PRIVATE);
+                final Map<String, ?> map = sp.getAll();
+                final JSONObject prefSon = new JSONObject();
+                for (final String key : map.keySet()) {
+                    if (includeKey(key)) {
+                        final Object value = map.get(key);
+                        if (
+                            (value instanceof Integer) ||
+                            (value instanceof Float) ||
+                            (value instanceof String) ||
+                            (value instanceof Boolean)
+                        ) {
+                            prefSon.put(key, value);
```

Comment:
Have you checked that the type is stored in the Json-Object or does the information be lost.

Reply:
The json correctly stores boolean string and number.